### ISSUE TITLE
PoC : Créer le modèle d'intelligence artificielle

### ISSUE THREAD

**Issue by zhishengtrieu** [cegepmatane/Analyseur-Immobilier#2]
# Description du fini : 
Le modèle doit d'abord être capable de reconnaitre quelle partie du bien est représenté dans l'image (chambres, cuisine, salle de bain, façade, etc…). Cela peut commencer avec le repérage des meubles ou autres éléments spécifiques à chaque partie du bien (lit, table, four, …)

## A Faire :
- [x] Mise en place d'un modèle CNN
- [x] Avoir un dataset d'images de chambres, cuisines, …

**Comment by zhishengtrieu** [comment]
Le modèle prédit constamment la même chose : 
```
1/1 [==============================] - 0s 240ms/step
[[0. 0. 1. 0.]]
frontal
```

**Comment by zhishengtrieu** [comment]
les prédictions arrive enfin a varier mais elle sont peu précises : 
```
2023-02-24 14:24:38.029894: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)
 to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
1/1 [==============================] - 0s 104ms/step
[[ 4.0034356e-06 -1.5995780e-04 -9.4237330e-05  4.6461901e-06]]
kitchen
```
L'accuracy (précision) du modèle est bloquée à 0.25 c'est qui est très bas

**Comment by zhishengtrieu** [comment]
Avec ce code :
```python

model = models.Sequential()
model.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(img_height, img_width, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(128, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(256, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Flatten())
model.add(layers.Dropout(0.5))
model.add(layers.Dense(128, activation='relu'))
model.add(layers.Dense(4, activation='softmax'))

model.summary()

print("Compilation du modèle")
model.compile(optimizer='adam',
              loss='categorical_crossentropy',
              metrics=['accuracy'])
```

On obtient une accuracy d'environ 0.85 : 

```
Nombre d'images chargées: 2140
Nombre de labels chargés: 2140
2023-02-24 15:40:43.168865: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network
 Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #
=================================================================
 conv2d (Conv2D)             (None, 62, 62, 32)        896

 max_pooling2d (MaxPooling2D  (None, 31, 31, 32)       0
 )

 conv2d_1 (Conv2D)           (None, 29, 29, 64)        18496

 max_pooling2d_1 (MaxPooling  (None, 14, 14, 64)       0
 2D)

 conv2d_2 (Conv2D)           (None, 12, 12, 128)       73856

 max_pooling2d_2 (MaxPooling  (None, 6, 6, 128)        0
 2D)

 conv2d_3 (Conv2D)           (None, 4, 4, 256)         295168

 max_pooling2d_3 (MaxPooling  (None, 2, 2, 256)        0
 2D)

 flatten (Flatten)           (None, 1024)              0

 dropout (Dropout)           (None, 1024)              0

 dense (Dense)               (None, 128)               131200

 dense_1 (Dense)             (None, 4)                 516

=================================================================
Total params: 520,132
Trainable params: 520,132
Non-trainable params: 0
_________________________________________________________________
Compilation du modèle
Entrainement du modèle
Epoch 1/10
51/51 [==============================] - 6s 93ms/step - loss: 3.5768 - accuracy: 0.4312 - val_loss: 0.9360 - val_accuracy: 0.5944
Epoch 2/10
51/51 [==============================] - 5s 90ms/step - loss: 0.9106 - accuracy: 0.5863 - val_loss: 0.7717 - val_accuracy: 0.6935
Epoch 3/10
51/51 [==============================] - 5s 91ms/step - loss: 0.7959 - accuracy: 0.6523 - val_loss: 0.7383 - val_accuracy: 0.6972
Epoch 4/10
51/51 [==============================] - 5s 92ms/step - loss: 0.7395 - accuracy: 0.6978 - val_loss: 0.8348 - val_accuracy: 0.6262
Epoch 5/10
51/51 [==============================] - 5s 91ms/step - loss: 0.6929 - accuracy: 0.7016 - val_loss: 0.6582 - val_accuracy: 0.7383
Epoch 6/10
51/51 [==============================] - 5s 91ms/step - loss: 0.6782 - accuracy: 0.7084 - val_loss: 0.5941 - val_accuracy: 0.7402
Epoch 7/10
51/51 [==============================] - 5s 92ms/step - loss: 0.5900 - accuracy: 0.7645 - val_loss: 0.5833 - val_accuracy: 0.7738
Epoch 8/10
51/51 [==============================] - 5s 92ms/step - loss: 0.4874 - accuracy: 0.8081 - val_loss: 0.5737 - val_accuracy: 0.7626
Epoch 9/10
51/51 [==============================] - 5s 95ms/step - loss: 0.4413 - accuracy: 0.8249 - val_loss: 0.6518 - val_accuracy: 0.7570
Epoch 10/10
51/51 [==============================] - 5s 94ms/step - loss: 0.4744 - accuracy: 0.8168 - val_loss: 0.6671 - val_accuracy: 0.7607
67/67 - 1s - loss: 0.3788 - accuracy: 0.8575 - 1s/epoch - 20ms/step
0.8574766516685486
```
